### ISSUE TITLE
Time zone inconsistency in dateTime* methods

### ISSUE THREAD

**Issue by bishopb** [fzaninotto/Faker#676]
There are four DateTime returning methods: `dateTime`, `dateTimeAD`, `dateTimeBetween`, and `dateTimeInterval`, Two of them set the time zone (`dateTimeBetween` and `dateTimeInterval`) on the returned object, while the other two do not. This leads to user land code having to juggle time zones when comparing dates returned from the different generators.

The generators should all return times in the same zone, unless the user specifically says to return times in different time zones.  PR #675 adds this support:

```
$faker->dateTime; // time zone is UTC
$faker->dateTime('now', 'America/New_York'); // custom time zone

// likewise for dateTimeAD, dateTimeBetween, and dateTimeInterval
```

**Comment by Newman101** [comment]
Right, I see what this is about. 

Method ```dateTimeBetween``` returns a timestamp, like this: 

```php
        $timestamp = mt_rand($startTimestamp, $endTimestamp);
        $ts = new \DateTime('@' . $timestamp);
        $ts->setTimezone(new \DateTimeZone(date_default_timezone_get()));
        return $ts;
```

Where as, ```dateTimeInInterval``` doesn't:

```php
return static::dateTimeBetween($begin, $end);
```

Question follows: should ```dateTimeBetween``` return such a time stamp as well? My opinion is that it actually should. 

Method ```dateTimeInterval``` gets a DateTime object based on a random date between two given dates, ```dateTimeInterval``` gets a DateTime object based on a random date between one given date and an interval.

**Comment by bishopb** [comment]
The now closed PR #675 standardized this behavior: all these methods now return a `\DateTime` with the timezone set to the system default, with the possibility to set the timezone in the generator.
